lms/djangoapps/verify_student/views.py

```diff
@@ -644,15 +652,14 @@ def _check_enrollment(self, user, course_key):
         return (has_paid, bool(is_active))
 
 
-def create_order_with_ecommerce_service(user, course_key, course_mode):     # pylint: disable=invalid-name
+def create_order_with_ecommerce_service(user, course_key, course_mode, processor):     # pylint: disable=invalid-name
     """ Create a new order using the E-Commerce API. """
     try:
         api = EcommerceAPI()
         # Make an API call to create the order and retrieve the results
-        _order_number, _order_status, data = api.create_order(user, course_mode.sku)
-
+        response_data = api.create_basket(user, course_mode.sku, processor)  # TODO get correct processor from request
         # Pass the payment parameters directly from the API response.
-        return HttpResponse(json.dumps(data['payment_parameters']), content_type='application/json')
+        return HttpResponse(json.dumps(response_data['payment_data'] or {}), content_type='application/json')
```

Comment:
Assuming this is meant to prevent exceptions from being raised if `payment_data` is missing from the response, this would be safer as `response_data.get('payment_data') or {}`.